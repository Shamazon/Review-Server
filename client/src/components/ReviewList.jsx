import ReviewListItem from './ReviewListItem.jsx'
import style from '../../dist/style/style.css'

const ReviewList = (props) => {
    return (
        <section>
            <h3>Top customer reviews</h3>
            {props.reviews.map((item, index) => <ReviewListItem review={item} key={index} />)}
        </section>
    )
}

export default ReviewList;