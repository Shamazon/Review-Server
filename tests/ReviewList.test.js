import React from 'react';
import ReviewList from '../client/src/components/ReviewList.jsx';
import enzyme from 'enzyme';

describe('ReviewList', () => {
  it('should be defined', () => {
    expect(ReviewList).toBeDefined();
  });
  it('should render correctly', () => {
    const props = {
      reviews: ['test'],
    };
    const tree = enzyme.shallow(<ReviewList {...props} />);
    expect(tree).toMatchSnapshot();
  });
});
