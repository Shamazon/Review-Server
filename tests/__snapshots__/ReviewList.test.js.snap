// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReviewList should render correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ReviewList
    reviews={
      Array [
        "test",
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h3>
          Top customer reviews
        </h3>,
        Array [
          <ReviewListItem
            review="test"
          />,
        ],
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Top customer reviews",
        },
        "ref": null,
        "rendered": "Top customer reviews",
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": "0",
        "nodeType": "function",
        "props": Object {
          "review": "test",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h3>
            Top customer reviews
          </h3>,
          Array [
            <ReviewListItem
              review="test"
            />,
          ],
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Top customer reviews",
          },
          "ref": null,
          "rendered": "Top customer reviews",
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "function",
          "props": Object {
            "review": "test",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
